From 12e24fdd5a8b8a63af953d9b380bce26da10b86f Mon Sep 17 00:00:00 2001
From: George Zacharia <george.zcharia@gmail.com>
Date: Wed, 2 Sep 2020 12:12:30 +0530
Subject: [PATCH] Add support for Russian Ruble and Turkish Lira

Test:
- Compiles,
- Switch to Turkish keyboard layout, observe Turkish Lira symbol on numeric key layout,
- Switch to Russian keyboard layout, observe Russian Ruble symbol on numeric key layout

Also clean up TODO statement pertaining to these changes

Change-Id: I6d90ecbbd5e8b0430fe65102656d34ba814c80c5
---
 java/res/xml/key_styles_currency.xml | 52 +++++++++++++++++++++++++---
 1 file changed, 47 insertions(+), 5 deletions(-)

diff --git a/java/res/xml/key_styles_currency.xml b/java/res/xml/key_styles_currency.xml
index 67b6d6be9..7e9a59e8b 100644
--- a/java/res/xml/key_styles_currency.xml
+++ b/java/res/xml/key_styles_currency.xml
@@ -93,7 +93,7 @@
              sr: Kosovo (sq_SR), Montenegro (sr_ME)
              sv: Sweden (sv_SV)
              tr: Cyprus (tr_CY) -->
-        <case latin:localeCode="bg|ca|cs|da|de|el|en|es|et|eu|fi|fr|ga|gl|hr|hu|it|lb|lt|lv|mt|nl|pl|pt|ro|sk|sl|sq|sr|sv|tr">
+        <case latin:localeCode="bg|ca|cs|da|de|el|en|es|et|eu|fi|fr|ga|gl|hr|hu|it|lb|lt|lv|mt|nl|pl|pt|ro|sk|sl|sq|sr|sv">
             <include latin:keyboardLayout="@xml/key_styles_currency_euro" />
         </case>
         <!-- ca: Catalan (Andorra, Spain)
@@ -113,12 +113,8 @@
              th: Thai (Baht)
              uk: Ukrainian (Hryvnia)
              vi: Vietnamese (Dong)  -->
-        <!-- TODO: The currency sign of Turkish Lira was created in 2012 and assigned U+20BA for
-             its unicode, although there is no font glyph for it as of November 2012. -->
         <!-- TODO: The currency sign of Armenian Dram was created in 2012 and assigned U+058F for
              its unicode, although there is no font glyph for it as of September 2013. -->
-        <!-- TODO: The currency sign of Russian Ruble was created in 2014 and assigned U+20BD for
-             its unicode, although there is no font glyph for it as of August 2014. -->
         <case latin:languageCode="fa|hi|iw|lo|mn|ne|th|uk|vi">
             <include latin:keyboardLayout="@xml/key_styles_currency_generic" />
         </case>
@@ -128,6 +124,29 @@
         <case latin:countryCode="LK|BD">
             <include latin:keyboardLayout="@xml/key_styles_currency_generic" />
         </case>
+        <case latin:localeCode="ru">
+            <!-- U+20BD: "₽" RUBLE SIGN
+                 U+00A3: "£" POUND SIGN
+                 U+20AC: "€" EURO SIGN
+                 U+00A2: "¢" CENT SIGN -->
+            <key-style
+                latin:styleName="currencyKeyStyle"
+                latin:keySpec="&#x20BD;"
+                latin:moreKeys="!text/morekeys_currency_generic" />
+            <key-style
+                latin:styleName="moreCurrency1KeyStyle"
+                latin:keySpec="&#x00A3;" />
+            <key-style
+                latin:styleName="moreCurrency2KeyStyle"
+                latin:keySpec="&#x20AC;" />
+            <key-style
+                latin:styleName="moreCurrency3KeyStyle"
+                latin:keySpec="$"
+                latin:moreKeys="&#x00A2;" />
+            <key-style
+                latin:styleName="moreCurrency4KeyStyle"
+                latin:keySpec="&#x00A2;" />
+        </case>
         <!-- bn_IN: Bengali (India) (Indian Rupee)
              en_IN: English (India) (Indian Rupee)
              kn_IN: Kannada (India) (Indian Rupee)
@@ -159,6 +178,29 @@
                 latin:styleName="moreCurrency4KeyStyle"
                 latin:keySpec="&#x00A2;" />
         </case>
+        <case latin:localeCode="tr">
+            <!-- U+20BA: "₺" TURKISH LIRA SIGN
+                 U+00A3: "£" POUND SIGN
+                 U+20AC: "€" EURO SIGN
+                 U+00A2: "¢" CENT SIGN -->
+            <key-style
+                latin:styleName="currencyKeyStyle"
+                latin:keySpec="&#x20BA;"
+                latin:moreKeys="!text/morekeys_currency_generic" />
+            <key-style
+                latin:styleName="moreCurrency1KeyStyle"
+                latin:keySpec="&#x00A3;" />
+            <key-style
+                latin:styleName="moreCurrency2KeyStyle"
+                latin:keySpec="&#x20AC;" />
+            <key-style
+                latin:styleName="moreCurrency3KeyStyle"
+                latin:keySpec="$"
+                latin:moreKeys="&#x00A2;" />
+            <key-style
+                latin:styleName="moreCurrency4KeyStyle"
+                latin:keySpec="&#x00A2;" />
+        </case>
         <!-- GB: United Kingdom (Pound) -->
         <case latin:countryCode="GB">
             <!-- U+00A3: "£" POUND SIGN
